File: Ejercicio 5/animales.java
interface AccionAnimal{
    void moverse();
}

abstract class Animal implements AccionAnimal{
    String nombre;

    Animal(String n){ nombre=n; }

    abstract void hacerSonido();
}

class Perro extends Animal implements AccionAnimal{
    Perro(String n){ super(n); }

    void hacerSonido(){ System.out.println(nombre+" ladra"); }
    public void moverse(){ System.out.println(nombre+" Corre y juega"); }
}

class Gato extends Animal implements AccionAnimal{
    Gato(String n){ super(n); }

    void hacerSonido(){ System.out.println(nombre+" maulla"); }
    public void moverse(){ System.out.println(nombre+" Camina sigilosamente"); }
}

class Vaca extends Animal implements AccionAnimal{
    Vaca(String n){ super(n); }

    void hacerSonido(){ System.out.println(nombre+" Hace muu"); }
    public void moverse(){ System.out.println(nombre+" Camina lentamente"); }
}

public class Main{
    public static void main(String[] args){

        Animal[] animales={
            new Perro("El perro de mi tia"),
            new Gato("Mi gato"),
            new Vaca("La vaca")
        };

        for(Animal a:animales){
            a.hacerSonido();
            a.moverse();
    }
}
}

File: Ejercicio 5/animales.py
from abc import ABC,abstractmethod

class AccionAnimal(ABC):
    @abstractmethod
    def moverse(self): pass

class Animal(ABC):
    def __init__(self,nombre):
        self.nombre=nombre

    @abstractmethod
    def hacerSonido(self): pass

class Perro(Animal,AccionAnimal):
    def hacerSonido(self): print(self.nombre," ladra")
    def moverse(self): print(self.nombre," Corre y juega")

class Gato(Animal,AccionAnimal):
    def hacerSonido(self): print(self.nombre," maulla")
    def moverse(self): print(self.nombre, " Camina sigilosamente")

class Vaca(Animal,AccionAnimal):
    def hacerSonido(self): print(self.nombre," hace muu")
    def moverse(self): print(self.nombre," Camina lentamente")

animales=[Perro("El perro de mi tia"),Gato("Mi gato"),Vaca("La vaca")]

for a in animales:
    a.hacerSonido()
    a.moverse()
